llvm/include/llvm/Analysis/FunctionPropertiesAnalysis.h

```diff
@@ -34,13 +34,13 @@ class FunctionPropertiesInfo {
   void reIncludeBB(const BasicBlock &BB);
 
   ir2vec::Embedding FunctionEmbedding = ir2vec::Embedding(0.0);
-  std::optional<ir2vec::Vocab> IR2VecVocab;
+  const ir2vec::Vocabulary *IR2VecVocab = nullptr;
```

Comment:
you can `const ir2vec::Vocabulary * const;` - the value is const, potentially `nullptr`, and also the field is set-once, right?

Reply:
The actual value is set in `FunctionPropertiesInfo::getFunctionPropertiesInfo`.

Reply:
If this isn't set in the constructor, how are you changing it from `nullptr`?

Reply:
Right. We optionally set Vocab in the factory method if IR2Vec is enabled. So, we cannot make the field const.